infoblox/resource_infoblox_a_record.go

```diff
@@ -291,15 +304,6 @@ func resourceARecordUpdate(d *schema.ResourceData, m interface{}) error {
 	connector := m.(ibclient.IBConnector)
 	objMgr := ibclient.NewObjectManager(connector, "Terraform", tenantID)
 
-	// Get the existing IP address
-	if ipAddr == "" && cidr == "" {
-		aRec, err := objMgr.GetARecordByRef(d.Id())
-		if err != nil {
-			return fmt.Errorf("failed getting A-record: %s", err.Error())
-		}
-		ipAddr = aRec.Ipv4Addr
-	}
```

Comment:
If we are removing this condition completely, we can add a condition for raising an error, if both parameters/fields are empty , something like this
```
if ipAddr == "" && cidr == "" {
  return fmt.Errorf("error updating A-record: only one of 'ip_addr' or 'cidr' is allowed to be non-empty")
}
```

Reply:
Ok